src/app/shared/layout/header.component.ts

```diff
@@ -67,7 +69,7 @@ export class HeaderComponent implements OnInit {
       });
 
     this.auth.currentAuthState.subscribe((state) => {
-      if (auth.isAuthenticated()) {
+      if (auth.isAuthenticated() && !cacheRepoFromUrlService.hasRepoLocation()) {
```

Comment:
Upon testing, I realised that changing repositories actually leads to having to authenticate again (as you mentioned in your comment). I think it might be good to persist session to sessionStorage in order for better UI/UX. However, I think this works pretty well by itself in the mean time, so let's add it as a separate issue and work on it after this is merged.

Reply:
I think gycgabriel mentioned above that this is an existing issue from CATcher. 
When we access pages by manually typing in a new url, it refreshes the whole app. Looking into this, I think we could save the github access token to sessionStorage to prevent re-authentication, but I'm not sure of how secure/safe this would be, considering the scope set for the accessToken